cmd/throttling.go

```diff
@@ -20,34 +20,60 @@ import (
 	"net/http"
 	"sync"
 	"time"
+
+	"github.com/minio/minio/cmd/config/api"
 )
 
-type apiThrottling struct {
-	mu      sync.RWMutex
-	enabled bool
+type apiConfig struct {
+	mu sync.RWMutex
 
 	requestsDeadline time.Duration
 	requestsPool     chan struct{}
+	readyDeadline    time.Duration
+	corsAllowOrigins []string
 }
 
-func (t *apiThrottling) init(max int, deadline time.Duration) {
-	if max <= 0 {
+func (t *apiConfig) init(cfg api.Config) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	t.readyDeadline = cfg.APIReadyDeadline
+	if cfg.APIRequestsMax <= 0 {
 		return
 	}
 
-	t.mu.Lock()
-	defer t.mu.Unlock()
+	apiRequestsMax := cfg.APIRequestsMax
+	if len(globalEndpoints.Hosts()) > 0 {
+		apiRequestsMax /= len(globalEndpoints.Hosts())
+	}
+
+	t.requestsPool = make(chan struct{}, apiRequestsMax)
+	t.requestsDeadline = cfg.APIRequestsDeadline
+}
+
+func (t *apiConfig) getCorsAllowOrigins() []string {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
+	return t.corsAllowOrigins
```

Comment:
You need to return a copy of the list, otherwise locking will be useless

Reply:
This is only one time @vadmeste in-fact we want to see a race if someone modifies it, copying would mean it will remove silent bugs.